From 6408ee42411f0d4153e7d4307f4325b4bd564b3b Mon Sep 17 00:00:00 2001
From: Jayachandran C <jayachandranc@google.com>
Date: Tue, 2 Sep 2025 05:59:26 +0000
Subject: [PATCH] Protect shell overriding the carrier config

Bug: 441823943
Flag: EXEMPT Bugfix
Test: Manually verified the scenario and all telephony and telecom CTS are verified

Merged-In: I2e5d5b59dd95aa60a07896bf335ebe14c97c411a
Change-Id: I2e5d5b59dd95aa60a07896bf335ebe14c97c411a
---
 src/com/android/phone/CarrierConfigLoader.java | 7 +++++++
 1 file changed, 7 insertions(+)

diff --git a/src/com/android/phone/CarrierConfigLoader.java b/src/com/android/phone/CarrierConfigLoader.java
index 307170a48..64efa9533 100644
--- a/src/com/android/phone/CarrierConfigLoader.java
+++ b/src/com/android/phone/CarrierConfigLoader.java
@@ -1322,6 +1322,13 @@ public class CarrierConfigLoader extends ICarrierConfigLoader.Stub {
         mContext.enforceCallingOrSelfPermission(
                 android.Manifest.permission.MODIFY_PHONE_STATE, null);
         //TODO: Also check for SHELL UID to restrict this method to testing only (b/131326259)
+
+        // Do not allow shell UID to override the carrier config. This will not impact
+        // the CTS and telephony shell commands as they use different uids
+        if (TelephonyPermissions.isShell(getCallingUid())) {
+            throw new SecurityException("overrideConfig cannot be invoked by shell");
+        }
+
         int phoneId = SubscriptionManager.getPhoneId(subscriptionId);
         if (!SubscriptionManager.isValidPhoneId(phoneId)) {
             logd("Ignore invalid phoneId: " + phoneId + " for subId: " + subscriptionId);
-- 
2.52.0.rc1.455.g30608eb744-goog

